### PR TITLE
Faster hybrid version of `sum(na.rm = FALSE)` when there are NA

### PR DESCRIPTION
**Pull request by romainfrancois**

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -17,7 +17,8 @@
 - new method `group_data()` (#3489).
 - joins no longer make lazy grouped data (#3566). 
 - new `nest_join()` function. `nest_join()` creates a list column of the matching rows. `nest_join()` + `tidyr::unnest()` is equivalent to `inner_join`  (#3570). 
-- `last_col()` is re-exported from tidyselect (#3584)
+- `last_col()` is re-exported from tidyselect (#3584). 
+- hybrid version of `sum(na.rm = FALSE)` exits early when there are missing values (#3288).
```
Comment (hadley) [inline NEWS.md]: I'd expand this by including something like: This considerably improves performance when there are missing values early in the vector, and then link to Jim's blogpost.

Comment (romainfrancois) [reply NEWS.md]: That does ring a 🔔 for when we played with naniar. @jimhester ?

Comment (jimhester) [reply NEWS.md]: It was https://gist.github.com/jimhester/96760387dec3a05898cd386f69e24b45

Comment (romainfrancois) [reply NEWS.md]: Thanks. I do remember that. Not really the same problem, and iirc @krlmlr has now fixed Rcpp’s is_na.

**Pull request merged**